File: Week 4/PRG_레벨2_요격 시스템/hyungjoon_solved.py
def solution(targets):
    answer = 0
    
    # e 구간 기준으로 정렬
    targets.sort(key=lambda x:x[1])
    
    # 요격 미사일의 현재 위치 초기화
    cur = -1
    # print(targets)
    
    for s, e in targets:
        # 요격 미사일의 현재위치(cur)가 폭격 미사일의 범위 외라면, 요격 미사일을 하나 더 발사한다.
        # 이 때, 이 미사일의 위치는 e - 0.5로 한다. (걸치는 구간에 쏘는건 요격을 못하니까)
        if cur < s:
            answer += 1
            cur = e - 0.5
    
    return answer

File: Week 4/PRG_레벨2_요격 시스템/hyungjoon_solved.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        double cur = -1;
        int answer = 0;

        // 끝나는 지점을 기준으로 오름차순 정렬
        Arrays.sort(targets, (a, b) -> {
            return a[1] - b[1];
        });

        // 기존에 발사한 요격 미사일(cur)로 커버가 안되는 미사일이라면 새로 하나 발사하고, 이 미사일의 범위를 e - 0.5로 지정
        // e - 0.5인 이유는 이 미사일로 뽕을 뽑기 위함임
        for(int i = 0; i < targets.length; i++){
            if(cur < targets[i][0]){
                answer++;
                cur = targets[i][1] - 0.5;
            }
        }

        return answer;
    }
}
